lib/errors.js

```diff
@@ -37,17 +37,42 @@ Errors.prototype = {
             line = line.line;
         }
 
-        // line and column numbers should be explicit
-        assert(typeof line === 'number' && line > 0,
-            'Unable to add an error, `line` should be a number greater than 0 but ' + line + ' given');
-        assert(typeof column === 'number' && column >= 0,
-            'Unable to add an error, `column` should be a positive number but ' + column + ' given');
-
-        this._addError({
+        var errorInfo = {
```

Comment:
Perhaps you mean `errorInfo` argument of the `cast` method? Specifically `additional` property.

In such case, yes, that would be preferable, but it has to be polymorphic since we don't know in what way rule would want apply the fix :-(